src/beta_inc.jl

```diff
@@ -0,0 +1,841 @@
+    elseif b == 0.0
+        return (0.0,1.0)
+    end
+
+    epps = max(eps(), 1.0e-15)
+    if max(a,b) < 1.0E-3 * epps
+        return (b/(a+b), a/(a+b))
+    end
+    ind = false
+    a0 = a
+    b0 = b
+    x0 = x
+    y0 = y
+    if min(a0,b0) > 1.0
+        @goto l30
+    end
+#PROCEDURE FOR A0<=1 OR B0<=1
+    if x > 0.5
+        ind = true
+        a0 = b
+        b0 = a
+        y0 = x
+        x0 = y
+
+    end
+    if b0 < min(epps, epps*a0)
+        println("fpser")
+        ans_p = fpser(a0,b0,x0,epps)
+        ans_q = 1.0 - ans_p
+        @goto l220
+    elseif a0 < min(epps, epps*b0)
+        println("apser")
+        ans_q = apser(a0,b0,x0,epps)
+        ans_p = 1.0 - ans_q
+        @goto l220
+    elseif max(a0,b0) > 1.0
+        if b0 <= 1.0 || (x0*b0)^a0 <= 0.7
+            @goto l100
+        elseif x0 >= 0.3
+            @goto l110
+        elseif x0 >= 0.1
+            if b0 > 15.0
+                @goto l131
+            else
+                n = 20
+                @goto l130
+            end
+        end
+    elseif a0 >= min(0.2, b0) || x0^a0 <= 0.9
+        @goto l100
+    elseif x0 >= 0.3
+        @goto l110
+    else
+        n = 20
+        @goto l130
+    end
+
+#PROCEDURE FOR A0>1 AND B0>1
+    @label l30
+     if a > b
+        lambda = (a+b)*y - b
+     else
+        lambda = a - (a+b)*x
+     end
+    if lambda < 0.0
+        ind = true
+        a0 = b
+        b0 = a
+        x0 = y
+        y0 = x
+        lambda = abs(lambda)
+    end
+    
+    if b0 < 40.0 && b0*x0 <= 0.7
+        @goto l100
+    elseif b0 < 40.0
+        n = trunc(Int, b0)
+        b0 -= float(n)
+        if b0 == 0.0
+            n-=1
+            b0=1.0
+        end
+        println("bup")
+        ans_p = bup(b0,a0,y0,x0,n,epps)
+        if x0 > 0.7
+            println("bup")
+            if a0 <= 15.0
+                n = 20
+                ans_p += bup(a0, b0, x0, y0, n, epps)
+                a0 += n
+            end
+        
+            println("bgrat")
+            ans_p = bgrat(a0,b0,x0,y0,15.0*eps())
+            ans_q = 1.0 - ans_p
+            @goto l220
+        end
+        println("bpser")
+        ans_p += bpser(a0,b0,x0,epps)
+        ans_q = 1.0 - ans_p
+        @goto l220
+    elseif a0 > b0
+        if b0 <= 100.0 || lambda > 0.03*b0
+            @goto l120
+        else
+            @goto l180
+        end
+    elseif a0 <= 100.0 || lambda > 0.03*a0
+        @goto l120
+    else
+        @goto l180
+    end
+
+#EVALUATION OF ALGOS FOR PROPER SUB-DOMAINS
+    @label l100
+     println("bpser")
+     ans_p = bpser(a0,b0,x0,epps)
+     ans_q = 1.0 - ans_p
+     @goto l220
+
+    @label l110
+     println("bpser")
+     ans_q = bpser(b0,a0,y0,epps)
+     ans_p = 1.0 - ans_q
+     @goto l220
+
+    @label l120
+     println("bfrac")
+     ans_p = bfrac(a0,b0,x0,y0,lambda,15.0*eps())
+     ans_q = 1.0 - ans_p
+     @goto l220
+
+    @label l130
+     println("bup")
+     ans_q = bup(b0,a0,y0,x0,n,epps)
+     b0 += n
```

Comment:
This doesn't look right: as it stands now, `ans_q` will be discarded. Should there be an `@goto l220`?

Reply:
No, the previous `ans_q` is passed as an argument to this if you see.